### PR TITLE
Fix 'play' command requiring 'offset' field

### LINKED ISSUE #1417: Connect: failed to handle request: unknown endpoint "play"
### Description
Trying to play a playlist on the Connect instance through Spotify web API outputs the following error in the logs and fails to start the playlist.

```
[2024-12-14T11:50:07Z ERROR librespot_connect::spirc] failed to handle request: Not implemented { unknown endpoint: Object {
        "context": Object {
            "uri": String("spotify:playlist:7Dkm7xoSC77lcMry2GvxIq"),
            "url": String("context://spotify:playlist:7Dkm7xoSC77lcMry2GvxIq"),
        },
        "endpoint": String("play"),
        "logging_params": Object {
            "device_identifier": String("webapi-4fe...a01"),
        },
        "options": Object {
            "seek_to": Number(0),
        },
        "play_origin": Object {
            "device_identifier": String("webapi-4fe...a01"),
        },
    } }
```

### Version
`0.6.0-dev 72b6ad9 (Built on 2024-12-10, Build ID: GUoggs4k, Profile: release)`

### How to reproduce
1. Launch `librespot` with 'librespot -j --verbose'\
2. Authenticate
3. Get matching device id
4. Call web api with that device id:
https://api.spotify.com/v1/me/player/play
```
{
    "context_uri": "spotify:album:5ht7ItJgpBH7W6vJ5BqpPr",
    "position_ms": 0
}
```
5. See mentioned error in log

### Log
```
[2024-12-14T12:10:49Z TRACE librespot_core::dealer::protocol] websocket request: Object {
        "command": Object {
            "context": Object {
                "uri": String("spotify:album:5ht7ItJgpBH7W6vJ5BqpPr"),
                "url": String("context://spotify:album:5ht7ItJgpBH7W6vJ5BqpPr"),
            },
            "endpoint": String("play"),
            "logging_params": Object {
                "device_identifier": String("webapi-cfe923b2d660439caf2b557b21f31221"),
            },
            "options": Object {
                "seek_to": Number(0),
            },
            "play_origin": Object {
                "device_identifier": String("webapi-cfe923b2d660439caf2b557b21f31221"),
            },
        },
        "message_id": Number(1158946744),
        "play_on_secondary_stream": Null,
        "sent_by_device_id": String("webapi-cfe923b2d660439caf2b557b21f31221"),
        "target_alias_id": Number(0),
    }
[2024-12-14T12:10:49Z DEBUG librespot_connect::spirc] handling: 'endpoint: unknown play' from webapi-cfe923b2d660439caf2b557b21f31221
[2024-12-14T12:10:49Z ERROR librespot_connect::spirc] failed to handle request: Not implemented { unknown endpoint: Object {
        "context": Object {
            "uri": String("spotify:album:5ht7ItJgpBH7W6vJ5BqpPr"),
            "url": String("context://spotify:album:5ht7ItJgpBH7W6vJ5BqpPr"),
        },
        "endpoint": String("play"),
        "logging_params": Object {
            "device_identifier": String("webapi-cfe923b2d660439caf2b557b21f31221"),
        },
        "options": Object {
            "seek_to": Number(0),
        },
        "play_origin": Object {
            "device_identifier": String("webapi-cfe923b2d660439caf2b557b21f31221"),
        },
    } }
```

### Host (what you are running `librespot` on):
- OS: Linux 
- Platform: x86

### Additional context
If I add the "offset" parameter, e.g. 
```
    "offset": {
        "position": 5
    },
```
it works fine. I'm guessing the request matching is too strict on this being always present.

### PR DESCRIPTION
**Pull request by FabioGNR**
Fixes #1417.

### CODE REVIEW

File: connect/src/model.rs [hunk 1]
```diff
@@ -23,19 +23,26 @@ pub enum PlayingTrack {
...
-        if let Some(uri) = value.track_uri {
-            PlayingTrack::Uri(uri)
-        } else if let Some(uid) = value.track_uid {
-            PlayingTrack::Uid(uid)
-        } else {
-            PlayingTrack::Index(value.track_index.unwrap_or_else(|| {
-                warn!("SkipTo didn't provided any point to skip to, falling back to index 0");
-                0
-            }))
+impl From<Option<SkipTo>> for PlayingTrack {
```
Comment (photovoltex) [inline connect/src/model.rs:26]: Can you instead of adjusting the `From`, derive `Default` for `SkipTo` and `unwrap_or_default`?

**Review by photovoltex — CHANGES_REQUESTED** [review]
Thanks for the find!

File: connect/src/model.rs (hunk 1 above)
Comment (FabioGNR) [reply connect/src/model.rs]: Thanks, that's nicer.

Comment (photovoltex) [reply connect/src/model.rs]: Yeah, you can easily run into deep indentation issues with rust. There are ways around it, but usually they are not the obvious one. Anyhow thanks for the adjustment^^

**Comment by roderickvd** [comment]
Perfect, thanks!

**Pull request merged**